src/graphql-client.ts

```diff
@@ -15,20 +15,13 @@ const link = new HttpLink({ uri, headers, fetch });
 
 export const client = new ApolloClient({ cache, link, defaultOptions: { query: { errorPolicy: "all" } } });
 
-type Mutation<T> = {
-    mutation: TypedDocumentNode<void, { input: T }>,
-    variables: { input: T },
-    asJson: () => unknown
-};
-
-export function createMutation<T>(name: keyof schema.Mutation, input: T): Mutation<T> {
-    return {
-        mutation: gql`mutation($input: ${name[0].toUpperCase() + name.slice(1)}Input!) {
-                        ${name}(input: $input) { __typename } }
-                     ` as TypedDocumentNode<void, { input: T }>,
-        variables: { input },
-        asJson: function() { return { ...this, mutation: print(this.mutation) } }
-    };
+export function createMutation<T>(name: keyof schema.Mutation, input: T) {
+    const mutation = Object.assign(
+        gql`mutation($input: ${name[0].toUpperCase() + name.slice(1)}Input!) {
+            ${name}(input: $input) { __typename } }
+        ` as TypedDocumentNode<void, { input: T }>,
+        { toJSON: () => print(mutation) });

```

Comment:
Two things:

1. Picky: the `${name}` and the `` ` as `` lines should be indented by two more spaces.
2. I slightly dislike using `Object.assign` and also the fact that the `toJSON` passes the object containing *itself* to `print`.  But it looks like going around it will be more work than its worth, so don't bother unless you see something obvious.

Reply:
Yeah I fiddled with that for a while before coming to the same conclusion and settling on this form. `Object.assign()` was a mistake though: spread is (minorly) better. I changed the indentation but I'm happy to change it again/back!